lib/filtrex.ex

```diff
@@ -42,6 +42,21 @@ defmodule Filtrex do
       do: parse_validated_structure(configs, valid_structured_map)
   end
 
+  @doc """
+  Encode filter into map
+  """
+  @spec encode(Filtrex.t) :: Map.t
+  def encode(%Filtrex{type: type, conditions: conditions, sub_filters: sub_filters}) do
+    %{
+      "filter" =>
+      %{
+        "type" => type,
+        "conditions" => Enum.map(conditions, &Filtrex.Condition.encode_condition/1),
+        "sub_filters" => Enum.map(sub_filters, &encode_sub_filters/1)
```

Comment:
Because `filtrex` elsewhere allows for an arbitrary level of recursion for `sub_filters`, let's make sure this is consistent with this encoding feature being added. Checkout [schema.json](https://github.com/rcdilorenzo/filtrex/blob/b8ae52f6b06e2bcac1104e75a4adac49792fa291/resources/schema.json) for the definition.